test/extended/networking/commatrix.go

```diff
@@ -0,0 +1,99 @@
+package networking
+
+import (
+	"context"
+	"fmt"
+	"os"
+	"path"
+	"path/filepath"
+
+	g "github.com/onsi/ginkgo/v2"
+	o "github.com/onsi/gomega"
+	"github.com/sirupsen/logrus"
+
+	"github.com/liornoy/node-comm-lib/commatrix"
+	clientutil "github.com/liornoy/node-comm-lib/pkg/client"
+	"github.com/liornoy/node-comm-lib/pkg/types"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+
+	exutil "github.com/openshift/origin/test/extended/util"
+	ssutil "github.com/openshift/origin/test/extended/util/ss"
+)
+
+var _ = g.Describe("[sig-network][Feature:commatrix][Serial]", func() {
+	oc := exutil.NewCLI("commatrix")
+
+	g.It("should cover all ports that the nodes are actually listening on", func() {
+		artifactsDir := filepath.Join(exutil.ArtifactDirPath(), "commatrix")
+		err := os.MkdirAll(artifactsDir, 0755)
+		o.Expect(err).NotTo(o.HaveOccurred())
+
+		g.By("generating the commatrix")
+		kubeconfig, ok := os.LookupEnv("KUBECONFIG")
+		if !ok {
+			g.Fail("must set the KUBECONFIG environment variable")
+		}
+
+		comMatrix, err := commatrix.New(kubeconfig, "", commatrix.AWS)
+		o.Expect(err).ToNot(o.HaveOccurred())
+
+		cs, err := clientutil.New(kubeconfig)
+		o.Expect(err).ToNot(o.HaveOccurred())
+
+		g.By("fetching open ports on the cluster")
+		nodesList, err := cs.Nodes().List(context.TODO(), metav1.ListOptions{})
+		o.Expect(err).ToNot(o.HaveOccurred())
+
+		tcpfile, err := os.OpenFile(path.Join(artifactsDir, "raw-ss-tcp"), os.O_APPEND|os.O_CREATE|os.O_WRONLY, 0644)
+		o.Expect(err).ToNot(o.HaveOccurred())
+		defer tcpfile.Close()
+
+		udpfile, err := os.OpenFile(path.Join(artifactsDir, "raw-ss-udp"), os.O_APPEND|os.O_CREATE|os.O_WRONLY, 0644)
+		o.Expect(err).ToNot(o.HaveOccurred())
+		defer tcpfile.Close()
+
+		nodesComDeatils := []types.ComDetails{}
+		for _, n := range nodesList.Items {
+			cds, err := ssutil.CreateComDetailsFromNode(oc, &n, tcpfile, udpfile)
```

Comment:
To be more clear on what this is doing let's consider changing to ss.ToComDetails or ss.ConvertToComDetails